app/models/user.rb

```diff
@@ -91,6 +92,10 @@ def can_preview_next_release?
     has_permission?(Permissions::PREVIEW_NEXT_RELEASE)
   end
 
+  def can_add_a_youtube_urls_to_promotional_features?
```

Comment:
@davidgisbey Sorry to reopen an old thread, but it's got me wondering: do we actually need a new permission for this specific feature?

I may just be out of the loop. But I was originally working under the assumption that there were already rules around who can/cannot update Promotional Features (e.g. the user has to be a GDS Admin). However now I've said it, I actually can't see anything to suggest the [Promotional Features](https://github.com/alphagov/whitehall/blob/186d8c50eb7e25f5b25ae0728e7fe02e70533fe3/app/controllers/admin/promotional_features_controller.rb) or [Promotional Feature Items](https://github.com/alphagov/whitehall/blob/186d8c50eb7e25f5b25ae0728e7fe02e70533fe3/app/controllers/admin/promotional_feature_items_controller.rb) controllers do enforce specific permissions. 😕

For example, the Organisations controller enforces permissions here:

https://github.com/alphagov/whitehall/blob/186d8c50eb7e25f5b25ae0728e7fe02e70533fe3/app/controllers/admin/organisations_controller.rb#L85-L92

with rules being defined here:

https://github.com/alphagov/whitehall/blob/186d8c50eb7e25f5b25ae0728e7fe02e70533fe3/lib/whitehall/authority/rules/organisation_rules.rb#L3-L16

and the fact that we don't also apply these rules to the Promotional Features is making me wonder if we have a bit of a loophole in our permissions model here? 🤔

Assuming that the URL/path for the Promotional Features page can be easily guessed _(spoiler: it can be)_, I think it means any authenticated user can change these regardless of their permissions or the organisation they belong to.

**TL;DR:** I don't think we need a user permission specifically to allow embedding YouTube videos. However I do think we're missing permission checks on these controllers in general.

Reply:
Yea we should do this. The permission is just to hide it for now so we can merge in. It'll be removed after.